ingestion-beam/src/main/java/com/mozilla/telemetry/metrics/KeyedCounter.java

```diff
@@ -0,0 +1,34 @@
+package com.mozilla.telemetry.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+
+/**
+ * Collection of counters separated by arbitrary keys.
+ */
+public class KeyedCounter {
+
+  static final Map<String, Counter> counters = new HashMap<>();
+
+  /**
+   * Increment the counter associated with the given key by 1.
+   */
+  public static void inc(String key) {
+    inc(key, 1);
+  }
+
+  /**
+   * Increment the counter associated with the given key by n.
+   */
+  public static void inc(String key, long n) {
+    getOrCreateCounter(key).inc(n);
+  }
+
+  @VisibleForTesting
+  static Counter getOrCreateCounter(String key) {
+    return counters.computeIfAbsent(key, k -> Metrics.counter(KeyedCounter.class, k));
```

Comment:
Can we pass in class to the KeyedCounter constructor, and then use that here, so that metrics get registered with the class they're being logically called from rather than always being registered to KeyedCounter?

Reply:
Consider this request out of scope now that I realize it's following the existing pattern from PerDoctypeCounter.